### DISCUSSION TITLE
need to shrink ceph-mon from 3 down to 1

### DISCUSSION THREAD

**Discussion by ghost** [rook/rook#11476] [Q&A]
I need to shrink ceph-mon from 3 down to 1 - how would I do it?

I've managed shrinking them down from 3 down to 2 by updating the CR and restarting ceph-operator, but it appears that I can't shrink 2 down to 1:

```
$ kubectl -n rook-ceph exec -it $(kubectl -n rook-ceph get pod -l "app=rook-ceph-tools" -o jsonpath='{.items[0].metadata.name}') -- ceph mon ok-to-stop c
Error EBUSY: not enough monitors would be available (a) after stopping mons [c]
command terminated with exit code 16
```

Any suggestions what would be the best way to do so?

P.S. I know that it is recommended to run 3 Mons in the Production, but in my case I need to shrink down the nodes from 3 to 1, hence am asking this.

**Comment by ghost** [comment]
As a workaround am currently using [`allowMultiplePerNode: true`](https://github.com/rook/rook/blob/v1.10.5/deploy/charts/rook-ceph-cluster/values.yaml#L105) switch which lets me scale down to a single node.

**Reply by travisn** [reply]
Is this just a temporary situation and you're adding more nodes? With a single node or mon there is just no redundancy and you could easily lose the cluster.

If you really want to scale down to a single mon, you would need to reset the mon quorum. There is a [Rook Krew plugin command](https://rook.io/docs/rook/latest/Troubleshooting/disaster-recovery/#restoring-mon-quorum) for this to `restore-quorum` to a single mon.

**Reply by ghost** [reply]
It is not a temporary situation, but a permanent one.
I know that there is no redundancy and this is okay, we are planning on sunsetting this host anyway.

Thank you for the command, I'll give it a shot.

Another question is that I am now running a single node, it still somehow setting 2 replicas to `csi-cephfsplugin-provisioner` & `csi-rbdplugin-provisioner` -- is it because I have 2 Mgr's?

```
$ kubectl -n rook-ceph get deployment
NAME                                                       READY   UP-TO-DATE   AVAILABLE   AGE
csi-cephfsplugin-provisioner                               1/2     2            1           98d
csi-rbdplugin-provisioner                                  1/2     2            1           98d
...
```

**Reply by ghost** [reply]
Hmm, I think there is a 2nd node still in CR. Gonna take a look tomorrow.

**Reply by ghost** [reply]
> Hmm, I think there is a 2nd node still in CR. Gonna take a look tomorrow.

Figured the replicas, setting this parameter resolved my "all-in-one" case. (it defaults to `2` when isn't set):
```
csi:
  provisionerReplicas: 1
```
